.circleci/config.yml

```diff
@@ -0,0 +1,306 @@
+          command: |
+            brew update
+            brew install erlang
+      - *build_package
+      - *test_package
+      - *store_package_artifacts
+
+  deploy_integration:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: integration
+    steps: *deploy_steps
+
+  deploy_dev1:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: dev1
+    steps: *deploy_steps
+
+  deploy_dev2:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: dev2
+    steps: *deploy_steps
+
+  deploy_uat:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: uat
+    steps: *deploy_steps
+
+# CircleCI skips a job for a tag by default.
+# A job must have a filters tags section to run as a part of a tag push
+# and all its transitively dependent jobs must also have a filters tags section.
+workflows:
+  version: 2
+  build_test_deploy:
+    jobs:
+      - build:
+          requires: ~
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - test:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - eunit:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - static_analysis:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - linux_package:
+          filters:
+            branches:
+              only:
+                - master
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - osx_package:
+          requires:
+            - test
+            - eunit
+            - static_analysis
+          filters:
+            branches:
+              only: master
+
+      - deploy_integration:
+          requires:
+            - test
+            - eunit
+            - static_analysis
+            - linux_package
+          filters:
+            branches:
+              only: master
+
+      - deploy_dev1:
+          requires:
+            - linux_package
+          filters:
+            branches:
+              only: env/dev1
+
+      - deploy_dev2:
+          requires:
+            - linux_package
+          filters:
+            branches:
+              only: env/dev2
+
+      - hodl:
+          type: approval
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*$/
+
+      - deploy_uat:
+          requires:
+            - linux_package
```

Comment:
How do we make sure that the package uploaded to GitHub is the same deployed to UAT?

Reply:
Currently it is not the same package in travis too? Or I'm missing something?